src/gsy_e/models/strategy/market_agents/two_sided_engine.py

```diff
@@ -47,6 +47,22 @@ def __repr__(self):
         return "<TwoSidedPayAsBidEngine [{s.owner.name}] {s.name} " \
                "{s.markets.source.time_slot:%H:%M}>".format(s=self)
 
+    def _update_requirements_prices(self, bid):
+        requirements = []
+        for requirement in bid.requirements or []:
+            updated_requirement = {**requirement}
+            if "price" in updated_requirement:
+                if "original_price" not in updated_requirement:
```

Comment:
Can you please explain why the `original_price` should be part of an requirement?

Reply:
I am sorry, but I think I also do not get the concept of a `price` requirement. Can you please explain when this could be useful?
When I post a bid like `Bid(energy=10, price=2)`, I already set the price, don't I? Why would I set an price requirement?

Reply:
Sure, let me elaborate.
The goal of the degrees of freedom is to allow buyers to have flexibility in the buying process, for instance I can post a bid `Bid(energy=10, price=2, requirements=[{"preferred_partners": ["Hannes"], "price": 1}])`

Reply:
So the bid in this case will be sold to "Hannes" for a lower price.
And as for the `original_price` requirement, it serves the same goal as the original_price attribute of the orders (denoting the price before adding/substracting grid fees)

Reply:
I see, thank you! So if the order was not forwarded yet, the "original_price" would not be part of the requirement, and from the moment of first forwarding, this member is in the requirement, right?
Do we have a validator for the requirements (I had a quick look but could not find it)? I mean to prohibit to let the user set the member "original_price" in order to make the grid fee calculation work. What do you think?

Reply:
> if the order was not forwarded yet, the "original_price" would not be part of the requirement, and from the moment of first forwarding, this member is in the requirement

That's correct!
Regarding the validator, I can now see its importance, thanks for foreseeing it!

Reply:
@hannesdiedrich the validator needs some thorough thinking and designing to cover all corner cases, if you agree I'll add this in follow up PR after merging both this and [the other](https://github.com/gridsingularity/gsy-e/pull/1375) PRs

Reply:
Ok, I am also fine with adding a ticket for this and postpone it a bit.

Reply:
@Hassan754 Please add a note into the ticket to tell us to consider some time for the `thorough thinking and designing to cover all corner cases` Thanks!

Reply:
> I see, thank you! So if the order was not forwarded yet, the "original_price" would not be part of the requirement, and from the moment of first forwarding, this member is in the requirement, right?

I think it might also be worth adding the `original_price` at the very beginning, before the order is forwarded. This might simplify our life and we could avoid needing to check every time if `"original_price"` is present or not.